src/environment.js

```diff
@@ -26,6 +37,66 @@ class Environment {
     return null;
   }
 
+  rawCommitData(commitSha) {
+    // Make sure commitSha is only alphanumeric characters to prevent command injection.
+    if (commitSha.length > 100 || !commitSha.match(/^[0-9a-zA-Z]+$/)) {
+      return '';
+    }
+
+    const child_process = require('child_process');
+    const format = GIT_FORMAT_LINES.join('%n'); // git show format uses %n for newlines.
+    const cmd = `git show ${commitSha} --quiet --format="${format}"`;
```

Comment:
Nice to have this so locked down. Commit sha goes through regex and format is a const. :+1: